src/app/containers/DashboardContainer.js

```diff
@@ -34,6 +46,46 @@ export const getDashboardPage = () => {
 };
 
 const DashboardContainer = () => {
+    const api = useApi();
+    const { createModal } = useModals();
+    const [plans, loadingPlans] = usePlans();
+    const [subscription] = useSubscription();
+    const isBlackFriday = useBlackFriday();
+    const checked = useRef(false);
+    const [user] = useUser();
+
+    const handleSelect = ({ planIDs = [], cycle, currency, couponCode }) => {
+        const plansMap = planIDs.reduce((acc, planID) => {
+            const { Name } = plans.find(({ ID }) => ID === planID);
+            acc[Name] = 1;
+            return acc;
+        }, Object.create(null));
```

Comment:
```
const plansMap = plans.reduce((acc, { ID, Name }) => {
    return planIDs.includes(ID) ? { ...acc, [Name]: 1 } : acc;
}, Object.create(null));
```

Reply:
it looks cleaner, but your approach is better if there are a lot more elements in `plans` than in `planIDs`.